peps/pep-0735.rst

```diff
@@ -0,0 +1,476 @@
+SHOULD prefer editable installs over non-editable installs. They MAY provide
+users with options to configure this behavior.
+
+Use of Dependency Groups
+------------------------
+
+Tools which support Dependency Groups are expected to provide new options and
+interfaces to allow users to install from Dependency Groups. Implementations
+may wish to treat them similarly to their current treatments of
+``requirements.txt`` files, or more similarly to extras.
+
+No syntax is defined for expressing the dependency group of a package, for two
+reasons:
+
+* it would not be valid to refer to the dependency groups of a third-party
+  package from pypi (because the data is defined to be unpublished)
+
+* there is not guaranteed to be a current package for dependency groups -- part
+  of their purpose is to support non-package projects
+
+For example, a possible ``pip`` interface for installing dependency groups
+would be:
+
+.. code:: shell
+
+    pip install --dependency-groups=test,typing
+
+Note that this is only an example. This PEP does not declare any requirements
+for how tools support the installation of Dependency Groups.
+
+Reference Implementation
+========================
+
+TODO! STUB!
+
+Planned reference implementation:
+A very simple "environment manager" which can be used to build virtualenvs from
+specified Dependency Groups. It will therefore need to fully support parsing of
+Dependency Groups.
+
+Backwards Compatibility
+=======================
+
+At time of writing, the ``requirements`` table is reserved for use by PEPs,
+meaning that this lays claim to a previously unused namespace.
+There should therefore be no direct backwards compatibility concerns.
+
+Future Compatiiblity
+--------------------
+
+This PEP defines explicit behaviors for tools when encountering unknown keys
+which must be followed in order to remain spec-compliant. These behaviors are
+meant to ensure that future PEPs can extend the data with very clear
+expectations about how existing tools will behave.
+
+Security Implications
+=====================
+
+This PEP introduces new syntaxes and data formats for specifying dependency
+information in projects. However, it does not introduce newly specified
+mechanisms for handling or resolving dependencies.
+
+It therefore does not carry security concerns other than those inherent in any
+tools which may already be used to install dependencies -- i.e. malicious
+dependencies may be specified here, just as they may be specified in
+``requirements.txt`` files.
+
+How to Teach This
+=================
+
+This feature should be referred to by its canonical name, "Dependency Groups".
+
+The basic form of usage should be taught as a variant on typical
+``requirements.txt`` data. PEP 508 package specifiers can be added to a named
+list. Rather than asking ``pip`` to install from a ``requirements.txt`` file,
+either ``pip`` or a relevant workflow tool will install from a named Dependency
```

Comment:
```suggestion
list. Rather than asking pip to install from a ``requirements.txt`` file,
either pip or a relevant workflow tool will install from a named Dependency
```